lib/core/src/obf.cpp

```diff
@@ -117,40 +119,33 @@ obfSetDebug( int opt ) {
 
 int
 obfiGetFilename( char *fileName ) {
-    char *envVar = NULL;
+    char* envVar{nullptr};
 
     envVar = getRodsEnvAuthFileName();
-    if ( envVar != NULL && *envVar != '\0' ) {
-        strcpy( fileName, envVar );
-        return 0;
-    }
-
-#ifdef windows_platform
-    if ( ProcessType != CLIENT_PT ) {
-        char *tmpstr1;
-        int t;
-        tmpstr1 = iRODSNtGetServerConfigPath();
-        sprintf( fileName, "%s\\irodsA.txt", tmpstr1 );
+    if (envVar != nullptr && *envVar != '\0') {
+        std::strcpy(fileName, envVar);
         return 0;
     }
 
-    /* for clients */
-    envVar = iRODSNt_gethome();
-#else
-    envVar =  getenv( "HOME" );
-#endif
-    if ( envVar == NULL ) {
+    envVar = std::getenv("HOME");
+    if (envVar == nullptr) {
         return ENVIRONMENT_VAR_HOME_NOT_DEFINED;
     }
-    strncpy( fileName, envVar, MAX_NAME_LEN );
-    strncat( fileName, "/", MAX_NAME_LEN );
-    strncat( fileName, AUTH_FILENAME_DEFAULT, MAX_NAME_LEN );
 
-#ifdef windows_platform
-    if ( envVar != NULL ) {
-        free( envVar );
+    using cstr = std::char_traits<char>;
+
+    const auto env_length_with_null_term{std::strlen(envVar) + 1};
+    constexpr auto auth_file_length{cstr::length(AUTH_FILENAME_DEFAULT)};
+    constexpr auto path_separator_length{1};
+
+    const auto required_size{env_length_with_null_term + auth_file_length + path_separator_length};
+    if (required_size > max_path_length) {
```

Comment:
Is `>` correct?
Does `max_path_length` consider the null byte?

Reply:
`max_path_length` should be the size of the entire buffer, so any size greater should not be able to fit into the existing buffer.

The additional null term for `required_size` is included in `env_length_with_null_term`.